src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -136,95 +187,30 @@ template <class T, class TRShape = result_shape_t<T>>
 std::vector<TRShape> shape_infer(const Squeeze* op,
                                  const std::vector<T>& input_shapes,
                                  const ITensorAccessor& ta = make_tensor_accessor()) {
-    using DimType = typename T::value_type;
-
-    const auto number_of_inputs = input_shapes.size();
-    OPENVINO_ASSERT(!input_shapes.empty());
-
     const auto& arg_shape = input_shapes[0];
     const auto& arg_rank = arg_shape.rank();
     auto output_shapes = std::vector<TRShape>(1);
     auto& output_shape = output_shapes[0];
 
-    std::unique_ptr<std::set<int64_t>> unique_axes;
+    std::unique_ptr<std::set<int64_t>> unique_axes{};
 
-    if (number_of_inputs == 1) {
-        unique_axes.reset(new std::set<int64_t>());
-    } else if (number_of_inputs == 2) {
-        const auto& axes_shape = input_shapes[1];
-        NODE_VALIDATION_CHECK(op,
-                              axes_shape.is_dynamic() || ov::util::is_rank_compatible_any_of(axes_shape.rank(), {0, 1}),
-                              "Second input (axes) should not be of rank higher than 1. Got: ",
-                              axes_shape.rank().get_length());
+    auto output_shape_for_squeezable_dim = [&]() {
```

Comment:
Why is lambda required to return only dynamic shape?
It looks like instead pass functor value can be used like `rank_modifier` which is 0 or 1.